### DISCUSSION TITLE
Can I add multiple services in parallel to my enclave via composition?

### DISCUSSION THREAD

**Discussion by guisellaDes** [kurtosis-tech/kurtosis#1276] [Q&A] [answered]
Can I add multiple services in parallel to my enclave via composition?

**Comment by leeederek** [comment, marked as answer]
Hey @guisellaDes!

Adding services in parallel is a great way to speed up how quickly your distributed system gets instantiated inside your enclave. By default, the [add_services](https://docs.kurtosis.com/starlark-reference/plan#add_services) instruction adds services in parallel with a default parallelism of 4 (which can be increased with the `--parallelism` flag).

However, when it comes to adding multiple services from different packages, you must do so within the `plan.add_services` instruction with the configuration for each service in a dictionary. You cannot currently import multiple packages (using locators) and run them in parallel without using the `plan.add_services` instruction because the call to run each of those imported packages starts the service itself.

As an example, if you have a service_a.star file that looks like this:
```python
def run()...

def get_config()...
```
Then you can add services from service_a.star in parallel into your main.star package with:
```python
a = import_module("./service_a.star")

def run():
   a_config = a.get_config()
   plan.add_services({"a": a_config})
```
